src/net.cpp

```diff
@@ -397,13 +447,13 @@ bool SeenLocal(const CService& addr)
 {
     {
         LOCK(cs_mapLocalHost);
-        if (mapLocalHost.count(addr) == 0)
+        if (fDebug10 || true) LogPrintf("SeenLocal %s", addr.ToString());
+        if (mapLocalHost.count(addr) == 0){
```

Comment:
addresses seen by peers are not added, but only increase the confidence in an address.
Addresses need to be added by other sources e.g. UPNP or from the Interfaces.
But if there are no addresses added, because UPNP is not used and no external addresses are found on the Intefraces, the address the peer told us is sent directly back to the peer, and no voting happens, thats somewhat inconsequent.